app/workers/twitter_worker.rb

```diff
@@ -18,19 +17,14 @@ def default_url_options
     }.new
   end
 
-  def perform(event_name, event_deadline, event_url, twitter_handle)
-    message_start = ["So great!","Hooray!","Amazing news!","Nice!!"]
-    message_end =
-    [
-      "is offering free diversity tickets! Apply before #{event_deadline} at #{event_url}!",
-      "just made free diversity tickets available! You can apply for them at #{event_url} until #{event_deadline}.",
-      "lets you apply for free diversity tickets! Submit your application before #{event_deadline} at #{event_url}!",
-      "has free diversity tickets available! Make sure to apply for them before #{event_deadline} here: #{event_url}!"
-    ]
-    message = [message_start.sample]
-    message << (twitter_handle ? "@#{twitter_handle}" : event_name.truncate(30, separator: ' '))
-    message << [message_end.sample]
+  def perform(event)
+    deadline = format_date(event.deadline)
+    event_url = routes.event_url(event)
+    name_or_handle = (event.twitter_handle ? "@#{event.twitter_handle}" : "#{event.name.truncate(30, separator: ' ')}")
+    message = ["So awesome! #{name_or_handle} is giving away #{event.number_of_tickets} #DiversityTickets — you can apply for them here: #{event_url} ",
+    "Are you from an underrepresented group and interested in #[TAG]? We have #{event.number_of_tickets} #DiversityTickets for #{name_or_handle} — apply by #{deadline} here: #{event_url}",
```

Comment:
We agreed to leave out the sentence with tags in it, because some events might not have assigned tags. we'll find a solution for that at a later point.
But we could use the rest without that first part like "Nice! We have [...] #DiversityTickets for [...] - apply by [...]"